packages/grafana-data/src/types/plugin.ts

```diff
@@ -16,6 +16,7 @@ export enum PluginType {
   datasource = 'datasource',
   app = 'app',
   renderer = 'renderer',
+  secretsmanager = 'secretsmanager',
```

Comment:
**Question:** how "temporary" is this?

I am asking because in case we introduce a new plugin type, then we should update the [`plugin.json` schema](https://github.com/grafana/grafana/blob/4eff846285f8b1fda07edccf1a19f5c5429d4398/docs/sources/developers/plugins/plugin.schema.json#L22) and also the [`plugin.json` docs](https://github.com/grafana/grafana/blob/b84294dd27386dd73669da07e2e8f70c77935e47/docs/sources/developers/plugins/metadata.md). I guess the difficulties here could be that in this case we would need to explain what the actual differences / benefits of using this plugin type are.

Although I generally don't believe in temporary solutions, I think as a worst case scenario we should probably add a comment here at least which explains that this one doesn't have any docs yet, and also a few words on what this new type is. (I am also not a fan of comments, but I guess devs will come across this new type and start looking in the docs to understand what this actually does, which can be confusing if all the information is only available in issues or discussions at this point).

Reply:
It's unofficially supported/internal currently so we don't want any docs. Similar to the renderer plugin.

Reply:
Right, as Marcus said we don't want to support community secrets plugins for the time being. When we officially launch this (aiming for 9.1) we will provide documentation on how to install and configure them, but not how to write them.